autoscaler/README.md

```diff
@@ -0,0 +1,37 @@
+# Horizontal Self Scaler container
+
+This container image watches over the number of schedulable nodes in the cluster and resizes
+the number of replicas in the required controller.
+
+Usage of pod_nanny:
+    --configmap <params>
+    --rc <replication-controller>
```

Comment:
kubectl makes it look easy - scale.go internally has implementations for each scalable resource.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/scale.go#L229

And then, RunScale uses all that code and makes it look easy.

```
// RunScale executes the scaling
func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, shortOutput bool, options *ScaleOptions) error {
        ...
	mapper, typer := f.Object(cmdutil.GetIncludeThirdPartyAPIs(cmd))
	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
		ContinueOnError().
		NamespaceParam(cmdNamespace).DefaultNamespace().
		FilenameParam(enforceNamespace, options.Recursive, options.Filenames...).
		ResourceTypeOrNameArgs(false, args...).
		Flatten().
		Do()
       ...
    if err := scaler.Scale(info.Namespace, info.Name, uint(count), precondition, retry, waitForReplicas); err != nil {
	       return err
    }
    ...
```